dirsrvtests/tests/suites/basic/basic_test.py

```diff
@@ -1506,7 +1507,287 @@ def test_suffix_case(topology_st):
     domain = Domain(topology_st.standalone, TEST_SUFFIX)
     assert domain.dn == TEST_SUFFIX
 
+
+def test_bind_disconnect_invalid_entry(topology_st):
+    """Test close connection on failed bind with invalid entry
+
+    :id: b378543e-32dc-432a-9756-ce318d6d654b
+    :setup: Standalone instance
+    :steps:
+        1: create/get user
+        2: bind and search as user
+        3: enable nsslapd-close-on-failed-bind attr
+        4: bind as non existing entry to trigger connection closure
+        5: verify connection has been closed and server is still running
+        6: cleanup
+    :expectedresults:
+        1: success
+        2: success
+        3: nsslapd-close-on-failed-bind attr set to on
+        4: returns INVALID_CREDENTIALS, triggering connection closure
+        5: success
+        6: success
+    """
+
+    INVALID_ENTRY="cn=foooo,%s" % DEFAULT_SUFFIX
+    inst = topology_st.standalone
+
+    dm = DirectoryManager(inst)
+
+    # create/get user
+    users = UserAccounts(inst, DEFAULT_SUFFIX)
+    try:
+        user = users.create_test_user()
+        user.set("userPassword", PW_DM)
+    except ldap.ALREADY_EXISTS:
+        user = users.get('test_user_1000')
+        pass
+
+    # verify user can bind and search
+    try:
+        inst.simple_bind_s(user.dn, PW_DM)
+    except ldap.LDAPError as e:
+        log.info('exception description: ' + e.args[0]['desc'])
+    try:
+        inst.search_s(DEFAULT_SUFFIX, ldap.SCOPE_SUBTREE, '(objectclass=top)', ['dn'])
+    except ldap.LDAPError as e:
+        log.info('exception description: ' + e.args[0]['desc'])
+
+    # enable and verify attr
+    try:
+        dm_conn = dm.bind()
+        dm_conn.config.replace('nsslapd-close-on-failed-bind', 'on')
+        assert (dm_conn.config.get_attr_val_utf8('nsslapd-close-on-failed-bind')) == 'on'
+    except ldap.LDAPError:
+        log.info('Failed to config replace')
+
+    # bind as non existing entry which triggers connection close
+    with pytest.raises(ldap.INVALID_CREDENTIALS):
+        inst.simple_bind_s(INVALID_ENTRY, PW_DM)
+
+    # verify the connection is closed but the server is still running
+    assert (inst.status())
+    with pytest.raises(ldap.SERVER_DOWN):
+        inst.search_s(DEFAULT_SUFFIX, ldap.SCOPE_SUBTREE, '(objectclass=top)', ['dn'])
+    try:
+        dm_conn = dm.bind()
+    except ldap.LDAPError:
+        log.info('dm bind failed')
+
+    # cleanup
+    try:
+        dm_conn.config.replace('nsslapd-close-on-failed-bind', 'off')
```

Comment:
If the test will fail before that point, the cleanup won't happen.
So please, use either a pytest fixture (for setup-cleanup), or either `finally:` block with `try:` block through the whole test.